spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerQueryLookupStrategy.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.repository.query;
+
+import java.lang.reflect.Method;
+
+import org.springframework.cloud.gcp.data.spanner.core.SpannerOperations;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerMappingContext;
+import org.springframework.data.projection.ProjectionFactory;
+import org.springframework.data.repository.core.NamedQueries;
+import org.springframework.data.repository.core.RepositoryMetadata;
+import org.springframework.data.repository.query.QueryLookupStrategy;
+import org.springframework.data.repository.query.QueryMethod;
+import org.springframework.data.repository.query.RepositoryQuery;
+import org.springframework.util.Assert;
+
+/**
+ * @author Balint Pato
+ * @author Chengyuan Zhao
+ */
+public class SpannerQueryLookupStrategy implements QueryLookupStrategy {
+
+	private final SpannerOperations spannerOperations;
+
+	private final SpannerMappingContext spannerMappingContext;
+
+	public SpannerQueryLookupStrategy(SpannerMappingContext spannerMappingContext,
+			SpannerOperations spannerOperations) {
+		Assert.notNull(spannerMappingContext,
+				"A valid SpannerMappingContext is required.");
+		Assert.notNull(spannerOperations, "A valid SpannerOperations is required.");
+		this.spannerMappingContext = spannerMappingContext;
+		this.spannerOperations = spannerOperations;
+	}
+
+	@Override
+	public RepositoryQuery resolveQuery(Method method, RepositoryMetadata metadata,
```

Comment:
not covered by tests, please cover it!

Reply:
I'm not so sure this one should be tested. Testing it would just mean ensuring that we're getting the same query method back as we pass in, which is testing based on definition instead of results. I think that the method's return type is the only non-re-writing-definition test that is possible.

Reply:
The resolution logic should be tested. As agreed verbally, we can address it when we have more query types to handle in the next PRs.